carma_wm_ctrl/include/carma_wm_ctrl/WMBroadcaster.h

```diff
@@ -381,6 +383,24 @@ class WMBroadcaster
   /*! \brief helps to populate upcoming_intersection_ids_ from local traffic lanelet ids
    */
   void publishLightId();
+  
+  /*!
+   * \brief Retrieve the vehicle ID from global vehicle parameters, and set instance memeber vehicle id
+   @param vehicle_id Vehicle ID from UniqueVehicleParams.yaml 
+   */
+  void setConfigVehicleId(std::string vehicle_id);
+
+   /*!
+   * \brief Sets the TCM Acknowledgement publish times. 
+   @param ack_pub_times the number of times it publishes TCM Acknowledgement 
+   */
+  void setConfigACKPubTimes(int ack_pub_times);
+
+  /*!
+  * \brief Construct TCM acknowledgement object and populate it with params. Publish the object for a configured number of times.
+  */
+  void pubTCMACK(j2735_msgs::Id64b tcm_req_id, uint16_t msgnum, uint8_t ack_status, const std::string ack_reason);
```

Comment:
string should be passed as const reference to prevent unneeded copy
const std::string& ack_reason